.github/workflows/beam_PostCommit_Java_IO_GCS_Performance_Tests.yml

```diff
@@ -0,0 +1,122 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PostCommit Java IO GCS Performance Tests
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['it/google-cloud-platform/**','.github/workflows/beam_PostCommit_Java_IO_GCS_Performance_Tests.yml']
+  issue_comment:
+    types: [created]
+  schedule:
+  - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_PostCommit_Java_IO_GCS_Performance_Tests:
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Java PostCommit IO GCS Performance Tests'
+#    runs-on: [self-hosted, ubuntu-20.04, main]
+    runs-on: ubuntu-20.04
+    name: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    strategy:
+      matrix:
+        job_name: ["beam_PostCommit_Java_IO_GCS_Performance_Tests"]
+        job_phrase: ["Run Java PostCommit IO GCS Performance Tests"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup repository
+      uses: ./.github/actions/setup-action
+      with:
+        comment_phrase: ${{ matrix.job_phrase }}
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    - name: Checkout release branch
+      if: github.event_name == 'schedule' #This has scheduled runs run against the latest release
+      uses: actions/checkout@v3
+      with:
+        ref: release-2.50.0 #TODO automate updating this
+        repository: apache/beam
+    - name: Checkout non-release branch
+      if: github.event_name != 'schedule' #This has triggered runs checkout the triggering branch
```

Comment:
You should probably get rid of this step and just do `if: github.event_name == 'schedule'` on the previous step. setup-action will get you the right branch in other cases

Reply:
done